src/controllerclientimpl.cpp

```diff
@@ -607,15 +607,33 @@ int ControllerClientImpl::_CallGet(const std::string& desturi, std::ostream& out
     CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_TIMEOUT_MS, 0L, (long)(timeout * 1000L));
     CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_HTTPHEADER, NULL, _httpheadersjson);
     CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_URL, NULL, desturi.c_str());
-    CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_WRITEFUNCTION, NULL, _WriteOStreamCallback);
-    CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_WRITEDATA, NULL, &outputStream);
+    _buffer.clear();
+    _buffer.str("");
+    struct scallback{
```

Comment:
can you define this class outside of the function scope?

Reply:
Does it mean that I make a new callback function as a ControllerClientImpl member? Well, sure...  Updated.

The very initial version used lambda (which VC2010 does not support casting to static function pointer, unfortunately), which was because I feel difficulty to give it a proper name.

Reply:
Also I found that the `tellg` needs to be `tellp` actually, so I fixed it as well.